wallet-core/src/profile.rs

```diff
@@ -12,14 +12,27 @@ use sha2::{Digest, Sha256};
 use execution_core::{
     signatures::bls::SecretKey as BlsSecretKey,
     transfer::phoenix::{
-        PublicKey as PhoenixPublicKey, SecretKey as PhoenixSecretKey,
+        Note, PublicKey as PhoenixPublicKey, SecretKey as PhoenixSecretKey,
         ViewKey as PhoenixViewKey,
     },
 };
 use zeroize::Zeroize;
 
 use crate::RNG_SEED;
 
+/// Calculate the sum for all the given [`Note`]s that belong to the given
+/// [`PhoenixViewKey`].
+///
+/// This function assumes that all notes are owned by the given view-key.
```

Comment:
This comment is somewhat misleading. It applies to transparent notes but not to obfuscated ones. The code does not assume successful decoding of an obfuscated note; instead, it defaults to `0` if decoding fails. The comment should be more comprehensive, clearly explaining the code’s behavior in both scenarios.

Reply:
Already removed 👍